File: gamslib/blend/blend-scalar.py
#  LP written by GAMS Convert at 12/13/18 10:24:28
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#          5        5        0        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         10       10        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         46       46        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x5 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)

m.obj = Objective(expr=   4.1*m.x1 + 4.3*m.x2 + 5.8*m.x3 + 6*m.x4 + 7.6*m.x5 + 7.5*m.x6 + 7.3*m.x7 + 6.9*m.x8 + 7.3*m.x9
                       , sense=minimize)

m.c1 = Constraint(expr=   10*m.x1 + 10*m.x2 + 40*m.x3 + 60*m.x4 + 30*m.x5 + 30*m.x6 + 30*m.x7 + 50*m.x8 + 20*m.x9 == 30)

m.c2 = Constraint(expr=   10*m.x1 + 30*m.x2 + 50*m.x3 + 30*m.x4 + 30*m.x5 + 40*m.x6 + 20*m.x7 + 40*m.x8 + 30*m.x9 == 30)

m.c3 = Constraint(expr=   80*m.x1 + 60*m.x2 + 10*m.x3 + 10*m.x4 + 40*m.x5 + 30*m.x6 + 50*m.x7 + 10*m.x8 + 50*m.x9 == 40)

m.c4 = Constraint(expr=   m.x1 + m.x2 + m.x3 + m.x4 + m.x5 + m.x6 + m.x7 + m.x8 + m.x9 == 1)

File: gamslib/blend/blend-scalar.jl
#  LP written by GAMS Convert at 10/11/20 12:15:21
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#          5        5        0        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         10       10        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         46       46        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x5, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)

@objective(m, Min, 4.1*x1 + 4.3*x2 + 5.8*x3 + 6*x4 + 7.6*x5 + 7.5*x6 + 7.3*x7 + 6.9*x8 + 7.3*x9)

@constraint(m, 10*x1 + 10*x2 + 40*x3 + 60*x4 + 30*x5 + 30*x6 + 30*x7 + 50*x8 + 20*x9 == 30)

@constraint(m, 10*x1 + 30*x2 + 50*x3 + 30*x4 + 30*x5 + 40*x6 + 20*x7 + 40*x8 + 30*x9 == 30)

@constraint(m, 80*x1 + 60*x2 + 10*x3 + 10*x4 + 40*x5 + 30*x6 + 50*x7 + 10*x8 + 50*x9 == 40)

@constraint(m, x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 == 1)
